dashboard/app/models/programming_expression.rb

```diff
@@ -16,9 +16,104 @@
 #  index_programming_expressions_on_programming_environment_id  (programming_environment_id)
 #
 class ProgrammingExpression < ApplicationRecord
+  include SerializedProperties
+
   belongs_to :programming_environment
   has_and_belongs_to_many :lessons, join_table: :lessons_programming_expressions
 
+  validates_uniqueness_of :key, scope: :programming_environment_id
+
+  serialized_attrs %w(
+    color
+  )
+
+  def self.properties_from_file(path, content)
+    expression_config = JSON.parse(content)
+
+    environment_name = File.basename(File.dirname(path)) == 'GamelabJr' ? 'spritelab' : File.basename(File.dirname(path))
+    programming_environment = ProgrammingEnvironment.find_by(name: environment_name)
+
+    if environment_name == 'spritelab'
+      {
+        key: expression_config['config']['docFunc'] || expression_config['config']['func'] || expression_config['config']['name'],
+        name: expression_config['config']['func'] || expression_config['config']['name'],
+        programming_environment_id: programming_environment.id,
+        category: expression_config['category'],
+        color: expression_config['config']['color']
+      }
+    else
+      {
+        key: expression_config['docFunc'] || expression_config['func'],
+        name: expression_config['func'],
+        programming_environment_id: programming_environment.id,
+        category: expression_config['category'],
+        color: ProgrammingExpression.get_category_color(expression_config['category'])
+      }
+    end
+  end
+
+  def self.get_category_color(category)
+    case category
+    # General
+    when 'Advanced'
+      '#19c3e1'
+      # Gamelab and Applab have different blue colors for this. Waiting to hear thoughts on what to do
```

Comment:
can we solve this by making this method take in both category and environment_name?

Reply:
The advanced category does not show in the toolbox. I think it would be good for someone from @code-dot-org/starlabs to weigh in on if we can actually just standardize to one color of blue here. The curriculum team did not seem very concerned about this since its not in the toolbox. https://codedotorg.slack.com/archives/C1B3PNDL7/p1614718268052000

Reply:
Also for context other categorys that are shared like Variables and Math the colors match